kernel/kernel-api/src/main/java/io/delta/kernel/internal/ScanImpl.java

```diff
@@ -291,4 +299,11 @@ private Set<String> loadPartitionColNames() {
         }
         return Collections.unmodifiableSet(partitionColumnNames);
     }
+
+    private StructType loadDataSchema() {
+        return new StructType(metadata.getSchema().fields().stream()
+            .filter(field ->
+                !partitionColumnNames.contains(field.getName().toLowerCase(Locale.ENGLISH)))
```

Comment:
why english? isn't it usally `Locale.ROOT`?

Reply:
Was just copying the above method switched them both